### PR TITLE
Simple ReST renderer for CLI help texts

### PR DESCRIPTION
**Pull request by happz**
Plugin documentation lives in plugin docstrings and help texts of its
fields. It is rendered with Sphinx to create nice HTML docs, and using
ReST directives improves context, readability and interlinking of those
HTML docs. But for CLI, tmt does not have a ready-to-use ReST renderer,
which means on CLI, directives recognized by ReST and Sphinx are messy,
poorly visible and create unpleasant experience.

Patch adds a very trivial ReST renderer which is applied to plugin
docstrings and help texts of their fields. Uses colors and formats some
basic structures like lists or `code-block`s.

Pull Request Checklist

* [x] implement the feature

## Previews

https://tmt--2574.org.readthedocs.build/en/2574/plugins/index.html

### CODE REVIEW

**Review by psss — APPROVED** [review]
Thanks, looks very nice! Could we perhaps keep colors for the notes and warnings consistent with how they are shown on the web? Proposing in f381882f together with a couple of minor changes.

**Comment by psss** [comment]
Two more things I've noticed: Seems that bullet points inside notes have an extra `NOTE` tag included in the text:

    NOTE: When testing ostree booted deployments tmt will use
    rpm-ostree as the package manager to perform the installation of
    requested packages. The current limitations of the rpm-ostree
    implementation are:

    * NOTE: Cannot install new version of already installed local rpm.
    * NOTE: No support for installing debuginfo packages at this time.

The `tests` field of the `discover.shell` plugin is missing a description.

**Comment by happz** [comment]
> Thanks, looks very nice! Could we perhaps keep colors for the notes and warnings consistent with how they are shown on the web? Proposing in [f381882](https://github.com/teemtee/tmt/commit/f381882f17da6e0e2ddc1d450f8a1027b52ffc73) together with a couple of minor changes.

Added a reply in https://github.com/teemtee/tmt/commit/f381882f17da6e0e2ddc1d450f8a1027b52ffc73#r138267199, the rest of changes look good to me.

**Comment by happz** [comment]
> Two more things I've noticed: Seems that bullet points inside notes have an extra `NOTE` tag included in the text:
> 
> ```
> NOTE: When testing ostree booted deployments tmt will use
> rpm-ostree as the package manager to perform the installation of
> requested packages. The current limitations of the rpm-ostree
> implementation are:
> 
> * NOTE: Cannot install new version of already installed local rpm.
> * NOTE: No support for installing debuginfo packages at this time.
> ```

Hopefully addressed in 2ee42e1d

> The `tests` field of the `discover.shell` plugin is missing a description.

True, and I would probably leave it for the later PR since the whole plugin docs require a review, and `tests` is just a drop.

**Review by janhavlin — APPROVED** [review]

**Review by lukaszachy — APPROVED** [review]

File: docs/templates/plugins.rst.j2 [hunk 1]
```diff
@@ -30,31 +28,58 @@
...
+{% set inherited_fields = container_inherited_fields(PLUGIN_DATA_CLASS) | sort %}
+{% set intrinsic_fields = container_intrinsic_fields(PLUGIN_DATA_CLASS) | sort %}
+
+{% for field_name in intrinsic_fields %}
+{{ render_field(field_name) }}
+{% endfor %}
+
+{% if inherited_fields %}
+Common Keys
+^^^^^^^^^^^
```
Comment (happz) [inline docs/templates/plugins.rst.j2:77]:
```suggestion
"""""""""""
```

**Pull request merged**